snappy/snapp.go

```diff
@@ -849,21 +849,21 @@ func (s *SnapPart) Install(inter progress.Meter, flags InstallFlags) (name strin
 		return "", err
 	}
 
-	// and finally make active
-	err = s.activate(inhibitHooks, inter)
-	defer func() {
-		if err != nil && oldPart != nil {
-			if cerr := oldPart.activate(inhibitHooks, inter); cerr != nil {
-				logger.Noticef("When setting old %s version back to active: %v", s.Name(), cerr)
+	if !inhibitHooks {
+		// and finally make active
+		err = s.activate(inhibitHooks, inter)
+		defer func() {
+			if err != nil && oldPart != nil {
+				if cerr := oldPart.activate(inhibitHooks, inter); cerr != nil {
```

Comment:
This isn't an atomic operation, which means there's a window of time between the two events that the system will be in a bad state. Again not a blocker for this branch as this was already there, but we need to encode this state persistently so that we can recover from that awkward place. Would be good to at least add a TODO here for the time being, and perhaps a Launchpad bug so we remember this is going to cause weird issues at some point.